### ISSUE TITLE
Nerf model is inconsistent with Paper.

### ISSUE THREAD

**Issue by jcao-ai** [keras-team/keras-io#1064]
1. According to section 3 of NeRF [paper](https://arxiv.org/abs/2003.08934):
> We encourage the representation to be multi-view consistent by restricting the network to predict the volumn density σ as a function of only the location x, while allowing the RGB color c to be predicted as a function of both location and viewing direction.

But the code seems to predict color and density simultaneously:
https://github.com/keras-team/keras-io/blob/d295b88f723dfc25970bd73e6c95fb9063b499ca/examples/vision/nerf.py#L384

2. Direction vector seems not being normalized to be unit vector.

Is this by design ? Thanks.

**Comment by sonali-kumari1** [comment]
Hi @jcao-ai - 
Thanks for pointing this out. The [3D volumetric rendering with NeRF](https://keras.io/examples/vision/nerf/) guide is the minimal implementation of the [NeRF paper](https://arxiv.org/abs/2003.08934) to provide an intuition of its core ideas and methodology. Predicting color and density simultaneously and not normalizing direction vector were most likely some simplifications made for clarity and conciseness.
